### PR TITLE
Feature/acp 2234/master automatic payment method domain registration for stripe connected accounts

### PR DESCRIPTION
**Pull request by stereomon**
- Developer(s): @stereomon

- Ticket: 

- Release Group: URL_HERE

- merge: merge

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   AppKernel             | minor (currently 0.x)  |                       |

-----------------------------------------

##### Change log

Improvements

- Removed logs for missing security manager plugin
- Load configuration from database before changing it.
- Refactored from single plugin to plugin stacks.

### CODE REVIEW

File: src/Spryker/Zed/AppKernel/Business/Reader/ConfigReader.php [hunk 1]
```diff
@@ -53,7 +51,6 @@ protected function configurePropelEncryption(AppConfigCriteriaTransfer $appConfi
         try {
             $this->propelEncryptionConfigurator->configurePropelEncryption($appConfigCriteriaTransfer->getTenantIdentifierOrFail());
         } catch (MissingSecretsManagerProviderPluginException) {
-            $this->getLogger()->warning(sprintf('There is no %s attached to %s::getSecretsManagerProviderPlugin(). This leads to unencrypted data in the database which should be avoided.', SecretsManagerProviderPluginInterface::class, SecretsManagerDependencyProvider::class));
```
Comment (matweew) [inline src/Spryker/Zed/AppKernel/Business/Reader/ConfigReader.php:56]: сс @pushokwhite

File: src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php [hunk 2]
```diff
@@ -70,21 +73,65 @@ public function saveConfig(AppConfigTransfer $appConfigTransfer): AppConfigRespo
...
-    protected function doSaveAppConfig(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
+    protected function mergeWithExistingConfiguration(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
     {
-        if ($this->configurationBeforeSavePlugin) {
-            $appConfigTransfer = $this->configurationBeforeSavePlugin->beforeSave($appConfigTransfer);
+        try {
+            /** @var \Generated\Shared\Transfer\AppConfigTransfer $existingAppConfigTransfer */
+            $existingAppConfigTransfer = $this->appKernelRepository->findAppConfigByCriteria(
+                (new AppConfigCriteriaTransfer())->setTenantIdentifier($appConfigTransfer->getTenantIdentifierOrFail()),
+                new AppConfigTransfer(),
```
Comment (matweew) [inline src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php:82]: `new AppConfigTransfer()`
why it's like this?  Seems it's a leftover, currently every app uses this transfer and it should not be provided as input.

File: src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php [hunk 3]
```diff
@@ -54,6 +55,8 @@ public function saveConfig(AppConfigTransfer $appConfigTransfer): AppConfigRespo
     {
         try {
             $appConfigTransfer = $this->getTransactionHandler()->handleTransaction(function () use ($appConfigTransfer) {
+                $appConfigTransfer = $this->mergeWithExistingConfiguration($appConfigTransfer);
```
Comment (matweew) [inline src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php]: it just sets the status
```suggestion
                $appConfigTransfer = $this->updateStatus($appConfigTransfer);
```

File: src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php (hunk 2 above)
Comment (stereomon) [reply src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php:82]: Can we re-evaluate this later?

File: src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php (hunk 3 above)
Comment (stereomon) [reply src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php]: The method name is open for anything else when we need it. Do you want to change anyway?

Comment (matweew) [reply src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php]: yes, YAGNI

+ it's just protected, you can always change it.

File: src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php (hunk 2 above)
Comment (matweew) [reply src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php:82]: but am I right?

Comment (stereomon) [reply src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php:82]: I think we could change it to always use but I need to double check it.

File: src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php [hunk 4]
```diff
@@ -70,21 +73,65 @@ public function saveConfig(AppConfigTransfer $appConfigTransfer): AppConfigRespo
      *
      * @return \Generated\Shared\Transfer\AppConfigTransfer
      */
-    protected function doSaveAppConfig(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
+    protected function mergeWithExistingConfiguration(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
```
Comment (stereomon) [inline src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php]:
```suggestion
    protected function updateStatus(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
```

**Review by matweew — APPROVED** [review]

**Pull request merged**